panos/policies.py

```diff
@@ -896,24 +909,118 @@ def _refresh_xml(self, elm):
+            count (int): Number of audit comments to return, maximum 5000.
+            direction (str): Specify whether logs are shown oldest first
+                (``forward``) or newest first (``backward``).
+            skip (int): Specify the number of logs to skip when doing log
+                retrieval.  This is useful when retrieving logs in batches
+                where you can skip the previously retrieved logs.
+
+        Returns:
+            list of :class:`panos.policies.AuditCommentLog`
+
+        """
+        dev = self.obj.nearest_pandevice()
+        xpath = self.obj.xpath()
+
+        # Build up the query.
+        query = "(subtype eq audit-comment)"
+        # Add in the name.
+        query += " and (path contains '\\'{0}\\'')".format(self.obj.uid)
+        # Add in the rule type.
+        query += " and (path contains '{0}')".format(
+            self.obj.xpath_short().split("/")[-2],
+        )
+        # Add in the rulebase type.
+        p = self.obj.parent
+        if p is None:
+            raise err.PanDeviceError("rule has empty parent")
+        if not any(isinstance(p, x) for x in (Rulebase, PreRulebase, PostRulebase)):
+            raise err.PanDeviceError("{0} has non-rulebase parent".format(self.obj.uid))
+        query += " and (path contains {0})".format(p.xpath().split("/")[-1])
+        # Add in the vsys or device group.
+        query += " and (path contains '\\'{0}\\'')".format(p.vsys)
+
+        extra_qs = {
+            "query": query,
+            "dir": direction,
+            "uniq": "yes",
+        }
+        if skip is not None:
+            extra_qs["skip"] = "{0}".format(int(skip))
+
+        resp = dev.xapi.log("config", count, extra_qs=extra_qs)
+
+        return [AuditCommentLog(x) for x in resp.findall("./result/log/logs/entry")]
+
+    def current(self):
+        """Returns the current audit comment.
+
+        Returns:
+            string
+
+        """
+        dev = self.obj.nearest_pandevice()
+        cmd = ET.Element("show")
+        sub = ET.SubElement(cmd, "config")
+        sub = ET.SubElement(sub, "list")
+        sub = ET.SubElement(sub, "audit-comments")
+        ET.SubElement(sub, "xpath").text = self.obj.xpath()
+
+        ans = self.obj.nearest_pandevice().op(
+            ET.tostring(cmd, encoding="utf-8"), cmd_xml=False,
+        )
+
+        resp = ans.find("./result/entry/comment")
+        if resp is not None:
+            return resp.text
+
+    def update(self, comment):
+        """Sets an audit comment for the given rule.
+
+        Args:
+            comment (str): The audit comment.
+
+        """
+        cmd = ET.Element("set")
+        sub = ET.SubElement(cmd, "audit-comment")
+        ET.SubElement(sub, "xpath").text = self.obj.xpath()
+        ET.SubElement(sub, "comment").text = comment
+
+        self.obj.nearest_pandevice().op(
+            ET.tostring(cmd, encoding="utf-8"), cmd_xml=False,
+        )
+
+
+class AuditCommentLog(OpStateObject):
```

Comment:
The classes for a lot of the op state stuff does do `Rule` and `Rulebase` prefixes, because those distinctions are important.  But there is no way to pull a rulebase audit comment history, so a `Rule` prefix is unnecessary.